src/Pelagos/Util/PubLinkUtil.php

```diff
@@ -34,37 +34,57 @@ public function __construct(EntityHandler $entityHandler)
      */
     public function fetchCitation($doi, $style = PublicationCitation::CITATION_STYLE_APA, $locale = 'en-US')
     {
-        $statusCodes = array(
-            200 => 'The request was OK.',
-            204 => 'The request was OK but there was no metadata available.',
-            404 => 'The DOI requested doesn\'t exist.',
-            406 => 'Can\'t serve any requested content type.',
-        );
+        $curlCitation = $this->curlCitation($doi);
+        $curlResponse = $curlCitation['curlResponse'];
+        $status = $curlCitation['status'];
+        $errorText = $curlCitation['errorText'];
 
+        $citation = new PublicationCitation($curlResponse, $style, $locale);
```

Comment:
Irrespective of the status, we are creating this object, but it is persisted only in the Controller based on the status returned. This util class also has a entityManager injected which is never used. Should we refactor this ?

Reply:
I agree, this function is doing way too much. Returning an entire set of status fields... It should just return data or an exception. I'm not sure if now is the time to refactor it?